### DISCUSSION TITLE
Slow channel switch & low upload speeds

### DISCUSSION THREAD

**Discussion by Sculas** [fabianishere/udm-iptv#87] [Q&A] [answered]
We (finally!) got KPN Fiber 1 Gbit today, goodbye Ziggo!

First we tested it with KPN's ExperiaBox (the modem you get with it), and everything ran smoothly with expected speeds of 1000 Mbps up and down.

Then we removed the ExperiaBox and followed every step and got everything setup, and it works!

There is one problem though, we're getting 950 Mbps download, which is fine, but something is limiting our upload speeds to 600 Mbps?! Does anyone know what this could be? This is a real bummer for us, since we're losing out on 400 Mbps (it's not free!).

Also, compared to when we tested IPTV directly to the ExperiaBox, channel switching seems a lot slower and takes around 3-5 seconds till the first frame.

What's going on here?

---- EDIT:

Both speedtests were done wired, with the laptop used directly connected to the UDM and ExperiaBox.

On the UDM, we have a separate VLAN (VLAN ID 4) for IPTV, where the only thing connected is the IPTV box.

The IPTV box is directly connected to the UDM, and we set the profile of the port to our IPTV VLAN.

Off-topic, but I would like to report that the instructions related to "DHCP Options" is outdated. It's now called "Custom DHCP Options" and the name "Vendor Class Identifier" is invalid because spaces aren't allowed.

Output of `udm-iptv-diag`:
```sh
=== Configuration ===
WAN Interface: eth4
WAN Ranges: 213.75.0.0/16 217.166.0.0/16
WAN VLAN: 4 (dev iptv)
WAN DHCP options: -O staticroutes -V IPTV_RG
LAN Interfaces: br4
IGMP Proxy quickleave disabled: false
IGMP Proxy debug: false
=== IP Link and Route ===
55: iptv@eth4: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue qlen 1000
    inet 10.204.56.132/22 brd 10.204.59.255 scope global iptv
       valid_lft forever preferred_lft forever
10.204.56.0/22 dev iptv  src 10.204.56.132 
213.75.112.0/21 via 10.204.56.1 dev iptv  metric 255 
=== Service Logs ===
-- Logs begin at Fri 2022-03-04 17:57:00 CET, end at Fri 2022-03-04 18:36:15 CET. --
Mar 04 18:01:51 ubnt systemd[1]: Started IPTV support for the UniFi Dream Machine.
Mar 04 18:01:51 ubnt udm-iptv[5688]: Obtaining IP address for VLAN interface
Mar 04 18:01:51 ubnt udm-iptv[5688]: udhcpc (v1.22.1) started
Mar 04 18:01:51 ubnt udm-iptv[5688]: Sending discover...
Mar 04 18:01:51 ubnt udm-iptv[5688]: Sending select for 10.204.56.132...
Mar 04 18:01:51 ubnt udm-iptv[5688]: Lease of 10.204.56.132 obtained, lease time 3600
Mar 04 18:01:51 ubnt udm-iptv[5688]: NATing IPTV network ranges (if necessary)
Mar 04 18:01:51 ubnt udm-iptv[5688]: Setting up igmpproxy
Mar 04 18:01:51 ubnt udm-iptv[5688]: Starting igmpproxy
```

---- EDIT 2:

Forgot to say, but disabling DPI and IPS did not seem to make a difference in upload speeds. Is that just a bottleneck of the UDM?
Smart Queues are off too.

**Comment by Sculas** [comment]
The low upload speeds seems to be a fluke or KPN issue, since it happens with their own modem too.

However, the slow channel switching is still a thing even after we'd reset the UDM to factory settings? Is that supposed to happen? When switching channels, it takes around 3 seconds for the first frame to show up, then it freezes and around 2 seconds later the video actually starts. Meaning switching channels takes around 5 seconds, doesn't sound normal to me?

**Comment by Sculas** [comment, marked as answer]
Well that took an unexpected turn. Seems like both are a KPN issue, or it's supposed to be.

**Reply by fabianishere** [reply]
Hmm interesting, you are seeing the same numbers with the Experiabox then?

**Reply by Sculas** [reply]
Eventually I tried again, comparing the UDM and Experiabox.

With iTV, there was little to no difference. Both took multiple seconds to finish switching channel.

With the Experiabox I was able to get 950/950 Mbps (up/down), but with the UDM I was only able to get 950/650-700 (up/down). Both tests were done wired with Speedtest by Ookla (Windows app). What was weird, is that when doing the speedtest on UniFi's dashboard, it gets 850/800-850 (up/down).

In the end, I'm confused. In any case, it's higher than what I previously had and it works. Though, it might be the UDM getting bottlenecked.

**Reply by fabianishere** [reply]
I suspect it might be related to the performance of the UDM’s PPPoE implementation or some kind of DPI (or other packet processing) happening.

**Reply by Sculas** [reply]
DPI (and IDS/IPS) is turned off (I double checked in the new and legacy interface) and this happened too right after resetting the UDM. So it might be their PPPoE implementation then, like you said.

**Comment by horolux** [comment]
Hi,

Do you have "Smart Queues" enabled on your WAN network?
